### PR TITLE
Lift data page

### PR DESCRIPTION
**Pull request by ManFromMons**

### CODE REVIEW

File: app/models/facilities_management/procurement_building_service.rb [hunk 1]
```diff
@@ -11,9 +11,11 @@ class ProcurementBuildingService < ApplicationRecord
     validates :size_of_external_area, numericality: { only_integer: true, greater_than: 0, message: :invalid }, allow_blank: true
     validates :no_of_consoles_to_be_serviced, numericality: { only_integer: true, greater_than: 0, message: :invalid }, allow_blank: true
     validates :tones_to_be_collected_and_removed, numericality: { only_integer: true, greater_than: 0, message: :invalid }, allow_blank: true
+    validates :lift_data, length: { minimum: 1 }, on: :lifts
+    validate :check_lift_data, on: :lifts
 
     REQUIRE_VOLUME_CODES = %w[E.4 G.1 G.3 G.5 K.1 K.2 K.3 K.7 K.4 K.5 K.6].freeze
-    SERVICES_AND_QUESTIONS = [{ code: 'C.5', questions: %i[total_floors_per_lift service_standard] },
```
Comment (iceraluk) [inline app/models/facilities_management/procurement_building_service.rb]: you should not remove service_standard here

File: config/routes.rb [hunk 2]
```diff
@@ -280,7 +281,7 @@
   end
 
   # scope module: :postcode do
-  #  resources :postcodes, only: :show
+  #  procurement_buildings_services :postcodes, only: :show
```
Comment (iceraluk) [inline config/routes.rb]: not sure what this is?

File: db/schema.rb [hunk 3]
```diff
@@ -20,7 +20,7 @@
   create_table "facilities_management_buildings", id: :uuid, default: nil, force: :cascade do |t|
     t.text "user_id", null: false
     t.jsonb "building_json", null: false
-    t.datetime "created_at"
+    t.datetime "created_at-retired"
```
Comment (iceraluk) [inline db/schema.rb]: not sure why this is here?

File: lib/tasks/error_pages.rake [hunk 4]
```diff
@@ -8,7 +8,7 @@ task :error_pages, :remote_site do |_t, args|
     sh "curl #{uri} --output #{html_file.path}"
     out_file = Tempfile.new('page-inline')
 
-    command = "npx juice --web-resources-relative-to #{site}/ --apply-style-tags false " \
+    command = "npx juice --web-procurement_buildings_services-relative-to #{site}/ --apply-style-tags false " \
```
Comment (iceraluk) [inline lib/tasks/error_pages.rake]: what is this for?

File: spec/controllers/facilities_management/beta/procurement_buildings_services_controller_spec.rb [hunk 5]
```diff
@@ -0,0 +1,10 @@
+require 'rails_helper'
+
+RSpec.describe FacilitiesManagement::Beta::ProcurementBuildingsServicesController, type: :controller do
+  describe 'GET #edit' do
+    it 'returns http success' do
+      get :show, params: { id: 9897987 }
```
Comment (iceraluk) [inline spec/controllers/facilities_management/beta/procurement_buildings_services_controller_spec.rb]: Some more tests for validations would be useful

File: app/models/facilities_management/procurement_building_service.rb (hunk 1 above)
Comment (ManFromMons) [reply app/models/facilities_management/procurement_building_service.rb]: Resolved this and the internal loop condition...

File: db/schema.rb (hunk 3 above)
Comment (ManFromMons) [reply db/schema.rb]: My bad - cheers for spotting this remnant

File: lib/tasks/error_pages.rake (hunk 4 above)
Comment (ManFromMons) [reply lib/tasks/error_pages.rake]: The generator created this - I will remove

File: config/routes.rb (hunk 2 above)
Comment (ManFromMons) [reply config/routes.rb]: This is another change I am not aware of - I have removed the comment block...

File: spec/controllers/facilities_management/beta/procurement_buildings_services_controller_spec.rb (hunk 5 above)
Comment (ManFromMons) [reply spec/controllers/facilities_management/beta/procurement_buildings_services_controller_spec.rb]: Hmmmm - I will add some

**Pull request merged**